server/types/oid/iterate.go

```diff
@@ -126,6 +138,14 @@ func IterateCurrentDatabase(ctx *sql.Context, callbacks Callbacks) error {
 			}
 		}
 	}
+
+	// Check for a type callback
+	if callbacks.Type != nil {
+		if err := iterateTypes(ctx, callbacks); err != nil {
+			return err
+		}
+	}
```

Comment:
Since we'll need to worry about schemas, this should also take into account the `SearchSchemas`. Perhaps if it's empty then we'll assume the default space. If it has a schema, then we'll check if it's `pg_catalog`.